src/common/storage_map.rs

```diff
@@ -1,11 +1,22 @@
+use crate::{binary, sparse, sum};
+
 use fuel_storage::Storage;
+use thiserror::Error;
 
 use std::borrow::Cow;
 use std::collections::HashMap;
-use thiserror::Error;
 
-#[derive(Clone, Debug, Error)]
-pub enum StorageError {}
+#[derive(Debug, Clone, Error)]
+pub enum StorageError {
```

Comment:
Originally, the purpose of this error enum was to encapsulate the errors thrown by `StorageMap`, the `Storage` trait impl. Really, the name should have been `StorageMapError` to better illustrate this. Because `StorageMap` is just a hash map, it doesn't actually have any errors, so this type was blank.

When defining a concrete Merkle tree type, you specify this enum as the error type: `SparseMerkleTree<StorageMapError>`, and this allows the Merkle tree to have knowledge about the error, but the `Storage` implementation and error type can be defined without knowledge of the rest of the `fuel-merkle` library. 

Generally speaking, the idea was that crates using `fuel-merkle` could define their own implementation of `fuel_storage::Storage` and a related error type outside and pass that to the tree definition, i.e. , `type = fuel_merkle::SparseMerkleTree<MyErrorType>`.

Reply:
> When defining a concrete Merkle tree type, you specify this enum as the error type: SparseMerkleTree<StorageMapError>, and this allows the Merkle tree to have knowledge about the error, but the Storage implementation and error type can be defined without knowledge of the rest of the fuel-merkle library.

It seems like would should change the Result type of the merkle tree methods to forward the generic `Error` type used by the storage impl. This way we wouldn't need anyhow:Error, and still retain a way to accomplish this goal.

Reply:
That sounds like a good solution. I wasn't familiar with the concept of forwarding errors at the time. If we can have a sort of `MerkleTreeError` that can be created from the generic storage impl error, that would solve this nicely.

Reply:
I created a separate issue to track this change here https://github.com/FuelLabs/fuel-merkle/issues/89.